### PR TITLE
Linkify the header of each notification group

### LINKED ISSUE #106: Linkify the header of each notification group
Greetings

I'd like to make a feature suggestion regarding the popup content:
**to linkify the header of each notification group**, 
i.e. linkify:

This is a very convenient 'Refined GitHub' extension feature:
> `linkify-notification-repository-header source, [screenshot](https://user-images.githubusercontent.com/1402241/80849887-81531c00-8c19-11ea-8777-7294ce318630.png)
> Linkifies the header of each notification group (when grouped by repository).

Thank you

### PR DESCRIPTION
**Pull request by darkred**
Resolves #106 

Based on how it's done in Refined GitHub: https://github.com/sindresorhus/refined-github/blob/master/source/features/linkify-notification-repository-header.tsx

Hope you like my attempt.

### CODE REVIEW

File: extension/github-notifications-preview.js [hunk 1]
```diff
@@ -145,6 +145,12 @@ async function openDropdown({currentTarget: indicator}) {
 			select.exists('.js-notifications-group', container)
 		);
 
+		let temp;
+		for (const header of select.all('.js-notifications-group h6')) {
+			temp = header.innerText.trim();
+			header.innerHTML = `<a class="text-inherit" href="${temp}">${temp}</a>`;
```
Comment (fregante) [inline extension/github-notifications-preview.js:151]: - The `href` is missing the initial `/` from the example
- Why `innerText`? The example used `textContent`
- This is slightly unsafe, we can't concatenate user strings into `innerHTML`, Mozilla will reject it
```suggestion
		for (const header of select.all('.js-notifications-group h6')) {
			const link = doma('<a class="text-inherit"></a>');
			link.href = '/' + header.textContent.trim();
			wrap(header.firstChild, link);
```

Using `wrap`: https://github.com/sindresorhus/refined-github/blob/1fa2a55aa1d09555a10725db29ef15cd8614adda/source/helpers/dom-utils.ts#L41

Comment (darkred) [reply extension/github-notifications-preview.js]: - I noticed that, but the produced links were ok.
- My mistake, sorry. After various failed attempts it missed my attention.
- Unfortunately, I'm not very experienced in JavaScript, and definitely not familiar with the various excellent utilities you have written/use in Refined GitHub, in this case `doma` and `wrap`, so I wrongly thought that a simplistic `innerHTML` was the only, unavoidable solution.

Your suggested code is perfect of course, and very educative. 
Thanks a lot for the feedback!

**Comment by darkred** [comment]
@fregante I just added the missing `wrap` function declaration.

It builds ok but there are now links now nor text at all:

here it should have `gorhill/uBlock`.
(There's no error in console)

**Comment by darkred** [comment]
@fregante ,  I believe it's ok ( it works fine now). What do you say?

**Comment by fregante** [comment]
> it's ok now

You just reverted it to what it was before, it's not ok for the reason I specified before

**Comment by darkred** [comment]
Excuse my previous attempts. I think it's eventually ok.

**Comment by fregante** [comment]
The problem was that `doma` returns a `DocumentFragment`, which is _emptied_ once it's inserted `after`. So when you `append` to the `wrapper`, you're actually appending to the `DocumentFragment`, which is not in the document.

Anyway, `doma.one()` was the right solution, sorry for the bad initial suggestion

**Comment by darkred** [comment]
No problem, thank you for your time and help.

**Pull request merged**